hydrolib/core/dflowfm/tim/models.py

```diff
@@ -0,0 +1,73 @@
+from pathlib import Path
+from typing import Callable, Dict, List
+
+from pydantic.class_validators import validator
+
+from hydrolib.core.basemodel import ModelSaveSettings, ParsableFileModel
+
+from .parser import TimParser
+from .serializer import TimSerializer, TimSerializerConfig
+
+
+class TimModel(ParsableFileModel):
+    """Class representing a tim (*.tim) file.
+
+    Attributes:
+        data: Dictionary with keys \"comment\" & time as float and value as List of floats".\n
+        serializer_config: TimSerializerConfig
+
+    Raises:
+        ValueError: If the time and values are not parsable to float, or when the amount of columns differs per timeseries.
+    """
+
+    serializer_config = TimSerializerConfig()
+
+    comments: List[str]
+    timeseries: Dict[float, List[float]]
+
+    @classmethod
+    def _ext(cls) -> str:
+        return ".tim"
+
+    @classmethod
+    def _filename(cls) -> str:
+        return "sample"
+
+    @classmethod
+    def _get_serializer(
+        cls,
+    ) -> Callable[[Path, Dict, TimSerializerConfig, ModelSaveSettings], None]:
+        return TimSerializer.serialize
+
+    @classmethod
+    def _get_parser(cls) -> Callable:
+        return TimParser.parse
+
+    @validator("timeseries")
+    @classmethod
+    def _timeseries_values(
+        cls, v: Dict[float, List[float]]
+    ) -> Dict[float, List[float]]:
+        """Validates if the amount of columns per timeseries match.
+
+        Args:
+            v (Dict[float, List[float]]): Value to validate, the timeseries in this case.
+
+        Raises:
+            ValueError: If the time and values are not parsable to float, or when the amount of columns differs per timeseries.
+
+        Returns:
+            Dict[float, List[float]: Validated timeseries.
+        """
+        firstlengthset = False
+        for time in v:
+            if not firstlengthset:
+                length = len(v[time])
+                firstlengthset = True
+                continue
+
+            if length != len(v[time]):
+                raise ValueError(f"Problem with values in timeseries, for time {time}")
+            length = len(v[time])
+
+        return v
```

Comment:
```suggestion
        if len(v) == 0:
            return v
        
        values_iterator = iter(v.values())
        n_columns = len(next(values_iterator))
        
        if n_columns == 0:
            raise ValueError('Time series cannot be empty.')
        
        if not all(len(values) == n_columns for values in values_iterator):
            raise ValueError('All time series rows should have the same number of columns.')
    
        return v
```
or (for more detailed error messages):
```suggestion
        if len(v) == 0:
            return v
    
        timeseries_iterator = iter(v.items())
        _, columns = next(timeseries_iterator)
        n_columns = len(columns)
        
        if n_columns == 0:
            raise ValueError('Time series cannot be empty.')
        
        for time, values in timeseries_iterator:
            if len(values) != n_columns:
                raise ValueError(f'Time {time}: Expected {n_columns} columns, but was {len(values)}')
        
        return v
```

Please pay attention to clear error messages: informing the user that there is a problem, but not why, is not informative.